bootloader/wscript

```diff
@@ -188,43 +192,64 @@ def set_lsb_compiler(ctx):
     ctx.env.append_value('LINKFLAGS', lsb_target_flag)
 
 
+def check_sizeof_pointer(ctx):
+
+    def check(type, expected):
+        # test code taken from autoconf resp. Scons: this is a pretty clever
+        # hack to find that a type is of a given size using only compilation.
+        # This speeds things up quite a bit compared to straightforward code
+        # actually running the code.
+        # Plus: This works cross :-)
+        fragment = '''
+        int main() {
+            static int test_array[1 - 2 * !(((long int) (sizeof(%s))) == %d)];
+            test_array[0] = 0;
+            return 0;
+        }''' % (type, expected)
+        return ctx.check_cc(fragment=fragment, execute=False, mandatory=False)
+
```

Comment:
You are right, the cast in my code it wrong - it is not needed at all since we are dealing with unsigned integers anyway. (In our case the cast does no harm, though, since both numbers are very small.) In autoconf this code is used as part of a binary search to determine arbitrary signed integers.

Never the less, this is not why the build fail. The code compiles fine, but linking fails. See [the test of your code without the cast](https://ci.appveyor.com/project/htgoebel/pyinstaller/build/1.0.166). More seriously IMO is that the compiler is not switched to the correct pointer-size even if `vcallvars.bat` is run.